### ISSUE TITLE
agnoster theme doesn't honor oh-my-zsh.hide-status

### ISSUE THREAD

**Issue by MikaelSmith** [robbyrussell/oh-my-zsh#4561]
For repositories where querying git for the prompt is very slow, it'd be nice to be able to disable it. The agnoster theme doesn't currently support a method to do this, but the standard appears to be
`git config --add oh-my-zsh.hide-status 1`.

**Comment by drasill** [comment]
I love agnoster, and as my $HOME is on a git repo, I really need this too :)

**Comment by jjdonov** [comment]
@MikaelSmith @drasill you can modify the them to honor the status, in agnoster.zsh-theme (or you custom copy) simply add an if statement:

```zsh
## Main prompt
build_prompt() {
  RETVAL=$?
  prompt_status
  prompt_virtualenv
  prompt_context
  prompt_dir
  if [[ "$(command git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]];then
    prompt_git
    prompt_hg
  fi
    prompt_end
}
```

**Comment by MikaelSmith** [comment]
My PR did something similar, but applying it for hg as well makes sense.

**Comment by benatkin** [comment]
`hide-dirty` speeds it up for me!

``` bash
# current repo
git config --add oh-my-zsh.hide-dirty 1
# global (I use this on my VirtualBox guest where I have vagrant synced folders)
git config --global --add oh-my-zsh.hide-dirty 1
```

**Comment by kjakapat** [comment]
In stead of completely disable prompt features. You can tweak your `agnoster.zsh-theme` to work specifically with your `git` repo by:

1. comment all lines that have `vcs_info`,
2. edit line in `prompt_git()` function from:
`ref="$vcs_info_msg_0_"` 
to 
`ref="$(git branch 2>/dev/null | grep -Po '(?<=\* ).*$')"` 

So, basically, you now disable all `vcs_info` actions and use a native `git` command instead to check a directory statuses. As a result, you still can see your useful git prompt and with a speed that as fast as you work in non-git directory.

**Comment by MikaelSmith** [comment]
That sounds like a useful approach. My PR is pretty stale, I'm tempted to close it. Is there still value to it?

**Comment by kjakapat** [comment]
Apart from suppress the feature, you can try [this](https://stackoverflow.com/questions/12765344/oh-my-zsh-slow-but-only-for-certain-git-repo/47606567#47606567).

**Comment by ExpandingMan** [comment]
Almost a year later and this still seems not to be working for me, any update on what's the best approach here?

I used to do this back about a year or so ago, and I had manually gone in and changed `build_prompt` in agnoster.  I'd really like not to have to do that so I can just use the plain old master of `oh-my-zsh` and not have to do anything to update.

**Comment by SupermeReturns** [comment]
@ExpandingMan There is still no official solution to this problem, I adopted this PR [make git config oh-my-zsh.hide-status work with the agnoster theme](https://github.com/robbyrussell/oh-my-zsh/pull/6362) which worked for me.

**Comment by mcornella** [comment]
Fixed in #6362.
